hathor/p2p/manager.py

```diff
@@ -580,7 +588,11 @@ def _update_whitelist_cb(self, body: bytes) -> None:
     def connect_to_if_not_connected(self, peer: UnverifiedPeer | PublicPeer, now: int) -> None:
         """ Attempts to connect if it is not connected to the peer.
         """
-        if not peer.info.entrypoints:
+        if not peer.info.entrypoints or (
```

Comment:
I didn't get it. What is this change doing?

Reply:
This code tries to remove from the list of peers the ones that disconnected from us and to which we surely won't be able to connect anymore.

Previously, to detect this, we only had to check whether they had no entrypoints.

But now, with IPv6, new cases could happen that wouldn't be caught:
- We don't have IPv6 enabled, and the peer only has IPv6 entrypoints (no IPv4)
- We have IPv4 disabled, and the peer only has IPv4 entrypoints (no IPv6)

It seemed like we would want to remove them from the list as well.

Does it make sense?

Reply:
Got it. I guess the following implementation could be more straightforward:

1. Create an enum of types (IPv4=1 and IPv6=2).
2. Create a list of supported addresses (IPv4, IPv6 ou IPv4 | IPv6).
3. Filter all valid addresses. If empty, skip it.

I guess it would prevent some of the most complex logic you are adding.

Reply:
> Got it. I guess the following implementation could be more straightforward:
> 
>     1. Create an enum of types (IPv4=1 and IPv6=2).
> 
>     2. Create a list of supported addresses (IPv4, IPv6 ou IPv4 | IPv6).
> 
>     3. Filter all valid addresses. If empty, skip it.
> 
> 
> I guess it would prevent some of the most complex logic you are adding.

This is what is done for `sync-v1` / `sync-v2`. I think it's a good strategy, but I'm OK doing the refactor in a new PR.

Reply:
Are you ok with doing this in another PR, @msbrogli ? 

Then I'll merge this one already

Reply:
Ok. Let's do it separately.